src/middleware/company.authentication.middleware.ts

```diff
@@ -5,18 +5,36 @@ import { CHS_URL, UPDATE_FILING_DATE_URL } from '../config';
 import { getApplicationData } from "../utils/application.data";
 import { ApplicationData } from "../model";
 import { EntityNumberKey } from "../model/data.types.model";
+import { getTransaction } from "../service/transaction.service";
 
-export const companyAuthentication = (req: Request, res: Response, next: NextFunction) => {
+export const companyAuthentication = async (req: Request, res: Response, next: NextFunction) => {
   try {
     logger.debugRequest(req, `Company Authentication Request`);
 
     const appData: ApplicationData = getApplicationData(req.session);
-    const entityNumber: string = appData?.[EntityNumberKey] as string;
+    let entityNumber: string = appData?.[EntityNumberKey] as string;
+    let returnURL: string = UPDATE_FILING_DATE_URL;
 
+    if (req.path.match(/resume/)) {
```

Comment:
would endWith("/resume") be better. I think we check for that elsewhere.

Reply:
yeah , thought the resume passed a link query parameter(/resume?link=L3VwZGF) but seems that is not sent across during the GET request so will use  path.endsWith(/${RESUME})